pkg/clusteragent/admission/mutate/agent_sidecar/util.go

```diff
@@ -8,111 +8,42 @@
 package agentsidecar
 
 import (
-	"github.com/DataDog/datadog-agent/pkg/config"
-	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"fmt"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
-	"os"
 )
 
-const (
-	agentSidecarContainerName = "datadog-agent-injected"
+// withEnvOverrides applies the extraEnv overrides to the container
```

Comment:
Might be good to rename this file to something more descriptive than `util.go`. Maybe `overrides.go` or similar.